browser/ui/views/frame/brave_browser_view.cc

```diff
@@ -1064,39 +1071,63 @@ void BraveBrowserView::UpdateWebViewRoundedCorners() {
   // contents and devtools.
   contents_container_->layer()->SetRoundedCornerRadius(corners);
 
-  // In addition to giving the contents container rounded corners, we also
-  // need to round the corners of the native view holder that displays the web
-  // contents.
+  const auto in_split_view_mode =
+      !!SplitViewBrowserData::FromBrowser(browser_.get());
+
+  auto update_corner_radius = [in_split_view_mode](
+                                  views::NativeViewHost* contents_holder,
+                                  views::NativeViewHost* devtools_holder,
+                                  DevToolsDockedPlacement devtools_placement,
+                                  gfx::RoundedCornersF corners) {
+    // In addition to giving the contents container rounded corners, we also
+    // need to round the corners of the native view holder that displays the web
+    // contents.
+
+    // Devtools lies underneath the contents webview. Round all four corners.
+    if (devtools_holder) {
+      devtools_holder->SetCornerRadii(corners);
+    }
 
-  // Devtools lies underneath the contents webview. Round all four corners.
-  if (devtools_web_view_->holder()) {
-    devtools_web_view_->holder()->SetCornerRadii(corners);
-  }
+    if (!in_split_view_mode) {
+      // In order to make the contents web view and devtools appear to be
+      // contained within a single rounded-corner view, square the contents
+      // webview corners that are adjacent to devtools.
+      // TODO(sko) We need to override
+      // BrowserView::GetDevToolsDockedPlacement(). It depends on coordinate of
+      // it but in split view mode, the calculation is not correct.
```

Comment:
@simonhong , that was related to this TODO() part. Disabled devtools placement code temporarily.